pkg/plugins/backendplugin/instrumentation/instrumentation.go

```diff
@@ -99,20 +113,25 @@ type Cfg struct {
 
 // InstrumentCollectMetrics instruments collectMetrics.
 func InstrumentCollectMetrics(ctx context.Context, req *backend.PluginContext, cfg Cfg, fn func() error) error {
-	return instrumentPluginRequest(ctx, cfg, req, "collectMetrics", fn)
+	return instrumentPluginRequest(ctx, cfg, req, EndpointCollectMetrics, fn)
 }
 
 // InstrumentCheckHealthRequest instruments checkHealth.
 func InstrumentCheckHealthRequest(ctx context.Context, req *backend.PluginContext, cfg Cfg, fn func() error) error {
-	return instrumentPluginRequest(ctx, cfg, req, "checkHealth", fn)
+	return instrumentPluginRequest(ctx, cfg, req, EndpointCheckHealth, fn)
 }
 
 // InstrumentCallResourceRequest instruments callResource.
-func InstrumentCallResourceRequest(ctx context.Context, req *backend.PluginContext, cfg Cfg, fn func() error) error {
-	return instrumentPluginRequest(ctx, cfg, req, "callResource", fn)
+func InstrumentCallResourceRequest(ctx context.Context, req *backend.PluginContext, cfg Cfg, requestSize float64, fn func() error) error {
+	pluginRequestSizeHistogram.WithLabelValues("backend-datasource", req.PluginID, EndpointCallResource,
+		string(cfg.Target)).Observe(requestSize)
+	return instrumentPluginRequest(ctx, cfg, req, EndpointCallResource, fn)
 }
 
 // InstrumentQueryDataRequest instruments success rate and latency of query data requests.
-func InstrumentQueryDataRequest(ctx context.Context, req *backend.PluginContext, cfg Cfg, fn func() error) error {
-	return instrumentPluginRequest(ctx, cfg, req, "queryData", fn)
+func InstrumentQueryDataRequest(ctx context.Context, req *backend.PluginContext, cfg Cfg,
+	requestSize float64, fn func() error) error {
+	pluginRequestSizeHistogram.WithLabelValues("backend-datasource", req.PluginID, EndpointQueryData,
```

Comment:
Regarding the `source` label, I have some concerns about this as mentioned here https://github.com/grafana/grafana/pull/65527#discussion_r1158922049

Reply:
you can name it `grafana-backend`